perkeep/perkeep
# server/perkeepd/ui: Drag-and-drop upload files and directories

# Summary:

Re-implement drag-and-drop handler to upload files and optionally support directories if browser implements `webkitGetAsEntry` on `DataTransferItem`.

# Test Plan:

Case A) Directory

1. Drag and drop a directory
2. Perkeep should receive all files, even deeply nested ones

Case B) Files

1. Drag and drop a selection of many files
2. Perkeep should receive all files

Case C) Other elements

1. Drag and drop non-file and non-directory elements such as an HTML image
2. Nothing should happen

# Closes: #54 #657
